webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js

```diff
@@ -0,0 +1,198 @@
+import { Table } from 'foremanReact/components/PF4/TableIndexPage/Table/Table';
+import { useBulkSelect } from 'foremanReact/components/PF4/TableIndexPage/Table/TableHooks';
+import { getPageStats } from 'foremanReact/components/PF4/TableIndexPage/Table/helpers';
+import { getControllerSearchProps, STATUS } from 'foremanReact/constants';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+
+import OpenscapRemediationWizardContext from '../OpenscapRemediationWizardContext';
+import WizardHeader from '../WizardHeader';
+import {
+  HOSTS_API_PATH,
+  HOSTS_API_REQUEST_KEY,
+  FAIL_RULE_SEARCH,
+} from '../constants';
+
+const ReviewHosts = () => {
+  const { source, hostId, setHostIdsParam } = useContext(
+    OpenscapRemediationWizardContext
+  );
+
+  const defaultSearch = `${FAIL_RULE_SEARCH} = ${source}`;
+  const defaultParams = {
+    search: defaultSearch,
+  };
+
+  const [params, setParams] = useState(defaultParams);
+
+  const searchProps = getControllerSearchProps('hosts');
+  searchProps.autocomplete.searchQuery = defaultSearch;
+  searchProps.autocomplete.url = '../../hosts/auto_complete_search'; // TODO: find a way to not use relative path
+  searchProps.disabled = true; // This is to force hosts to be searched by currently remediated rule only
+
+  const response = useAPI('get', `${HOSTS_API_PATH}?include_permissions=true`, {
+    key: HOSTS_API_REQUEST_KEY,
+    params: defaultParams,
+  });
+  const {
+    response: {
+      search: apiSearchQuery,
+      results,
+      per_page: perPage,
+      page,
+      subtotal,
+      message: errorMessage,
+    },
+    status = STATUS.PENDING,
+    setAPIOptions,
+  } = response;
+
+  const subtotalCount = Number(subtotal ?? 0);
+
+  const setParamsAndAPI = newParams => {
+    setParams(newParams);
+    setAPIOptions({ key: HOSTS_API_REQUEST_KEY, params: newParams });
+  };
+
+  const onSearch = newSearch => {
+    if (newSearch !== apiSearchQuery) {
+      setParamsAndAPI({ ...params, search: newSearch, page: 1 });
+    }
+  };
+
+  const { pageRowCount } = getPageStats({
+    total: subtotalCount,
+    page,
+    perPage,
+  });
+  const { fetchBulkParams, ...selectAllOptions } = useBulkSelect({
+    results,
+    metadata: { total: subtotalCount, page },
+    initialSearchQuery: apiSearchQuery || defaultSearch,
+    isSelectable: () => true,
+    initialArry: [hostId],
+  });
+  const {
+    selectPage,
+    selectedCount,
+    selectOne,
+    areAllRowsOnPageSelected,
+    areAllRowsSelected,
+    isSelected,
+  } = selectAllOptions;
+
+  useEffect(() => {
+    if (selectedCount) setHostIdsParam(fetchBulkParams());
+  }, [selectedCount, fetchBulkParams, setHostIdsParam]);
+
+  const selectionToolbar = (
+    <ToolbarItem key="selectAll">
+      <SelectAllCheckbox
+        {...{
+          selectAll: noop, // I don't think it really can select all since ids from other pages are still need to be loaded/fetched
```

Comment:
We can send a search query to the job invocation maybe?

Reply:
or just have selectAll: selectPage instead, as now the button doesn't select anything and its confusing

Reply:
Hmm... Select All option should not be available if `noop` is passed: https://github.com/theforeman/foreman/blob/181e5040e6026caa74daa2f30809ceba7dc80bf0/webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/Table/SelectAllCheckbox/index.js#L27